modules/standard/Map.chpl

```diff
@@ -378,20 +395,6 @@ module Map {
 
       :returns: Reference to the value mapped to the given key.
     */
-    proc ref this(k: keyType) ref
```

Comment:
We might have had this be the one defined because it actually generates something that can be linked to, where the other version is currently unlinkable (which we probably want to investigate)

Reply:
Ah yes that does seem to be case. A quick look indicates to me that its a combination of a return intent and `throws`.

I can revert this and put a TODO.

I don't believe there is an option issue about this, but I'll open one and make a note about this function in particular so the TODO can be fixed when the bug is.

Reply:
Issue opened here: #23776